synapseutils/copy.py

```diff
@@ -206,26 +248,17 @@ def _copyFile(syn, entity, destinationId, version=None, updateExisting=False, se
     bundle = syn._getEntityBundle(ent.id, version=ent.versionNumber, bitFlags=0x800|0x1)
     fileHandle = synapseclient.utils.find_data_file_handle(bundle)
     createdBy = fileHandle['createdBy']
-    #CHECK: If the user created the file, copy the file by using fileHandleId else hard copy
+    #CHECK: If the user created the file, copy the file by using fileHandleId else copy the fileHandle
     if profile.ownerId == createdBy:
-        new_ent = File(name=ent.name, parentId=destinationId)
-        new_ent.dataFileHandleId = ent.dataFileHandleId
+        newdataFileHandleId = ent.dataFileHandleId
     else:
-        #CHECK: If the synapse entity is an external URL, change path and store
-        if fileHandle['concreteType'] == 'org.sagebionetworks.repo.model.file.ExternalFileHandle':
-            store = False
-            path = ent.externalURL
-        else:
-            #####If you have never downloaded the file before, the path is None
-            store = True
-            #This needs to be here, because if the file has never been downloaded before
-            #there wont be a ent.path
-            ent = syn.get(entity,downloadFile=store,version=version)
-            path = ent.path
-
-        new_ent = File(path, name=ent.name, parentId=destinationId, synapseStore=store)
+        copiedFileHandle = copyFileHandles(syn, [fileHandle], ["FileEntity"], [bundle['entity']['id']])
+        newdataFileHandleId = copiedFileHandle['copyResults'][0]['newFileHandle']['id']
+
```

Comment:
Where is the error checking or recovery of failed copyrequests.  The copiedFileHandle is no actually a fileHandle but copyResult that could have failures.  These need to be handled or raised.